### PR TITLE
misc: plumb context throughout cilium codebase

### PR DESCRIPTION
**Pull request by ianvernon**
Read per-commit.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9523)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Only failure is BPF NodePort with VXLAN (known flake)

**Comment by ianvernon** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: cilium/cmd/kvstore_delete.go [hunk 1]
```diff
@@ -27,18 +28,21 @@ var kvstoreDeleteCmd = &cobra.Command{
 	Short:   "Delete a key",
 	Example: "cilium kvstore delete --recursive foo",
 	Run: func(cmd *cobra.Command, args []string) {
-		setupKvstore()
-
 		if len(args) < 1 {
 			Fatalf("Please specify a key or key prefix to delete")
 		}
 
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```
Comment (joestringer) [inline cilium/cmd/kvstore_delete.go:35]: Should we add a default timeout const variable for the `30*time.Second` for all of this CLI?

File: pkg/kvstore/allocator/allocator.go [hunk 2]
```diff
@@ -143,7 +143,6 @@ func (k *kvstoreBackend) lockPath(ctx context.Context, key string) (*kvstore.Loc
 // DeleteAllKeys will delete all keys
 func (k *kvstoreBackend) DeleteAllKeys() {
 	kvstore.DeletePrefix(context.TODO(), k.basePrefix)
-	kvstore.DeletePrefix(context.TODO(), k.basePrefix)
```
Comment (joestringer) [inline pkg/kvstore/allocator/allocator.go:146]: :thinking:

**Review by joestringer — APPROVED** [review]
Minor nit.

File: pkg/kvstore/allocator/allocator.go (hunk 2 above)
Comment (ianvernon) [reply pkg/kvstore/allocator/allocator.go:146]: This was added by me in a recent PR, so it does not affect any release branches ;)

File: cilium/cmd/kvstore_delete.go (hunk 1 above)
Comment (borkmann) [reply cilium/cmd/kvstore_delete.go:35]: @ianvernon please follow-up with @joestringer's feedback above in a separate PR, thanks.

**Pull request merged**